From e15b32b4e5274111de7540ac8f2d9246cc53ed52 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Tue, 24 Jun 2025 12:11:28 +0800
Subject: [PATCH 25/63] loonggpu: NFC: Clean up gsgpu_bo_move

Just use goto to deduplicate some identical code, and wrap a long line
to make the flow more similar to the reference implementation.

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 loonggpu/loonggpu_ttm.c | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/loonggpu/loonggpu_ttm.c b/loonggpu/loonggpu_ttm.c
index 8ba6a92..783ba2a 100644
--- a/loonggpu/loonggpu_ttm.c
+++ b/loonggpu/loonggpu_ttm.c
@@ -637,11 +637,10 @@ static int loonggpu_bo_move(struct ttm_buffer_object *bo, bool evict,
 	abo = ttm_to_loonggpu_bo(bo);
 	adev = loonggpu_ttm_adev(bo->bdev);
 
-	if (!old_mem || (old_mem->mem_type == TTM_PL_SYSTEM && bo->ttm == NULL)) {
+	if (!old_mem || (old_mem->mem_type == TTM_PL_SYSTEM
+			 && bo->ttm == NULL)) {
 		loonggpu_move_null(bo, new_mem);
-		atomic64_add(lg_gbo_to_gem_obj(abo).size, &adev->num_bytes_moved);
-		loonggpu_bo_move_notify(bo, evict, new_mem);
-		return 0;
+		goto out;
 	}
 
 	if ((old_mem->mem_type == TTM_PL_TT &&
@@ -650,9 +649,7 @@ static int loonggpu_bo_move(struct ttm_buffer_object *bo, bool evict,
 	     new_mem->mem_type == TTM_PL_TT)) {
 		/* bind is enough */
 		loonggpu_move_null(bo, new_mem);
-		atomic64_add(lg_gbo_to_gem_obj(abo).size, &adev->num_bytes_moved);
-		loonggpu_bo_move_notify(bo, evict, new_mem);
-		return 0;
+		goto out;
 	}
 
 	if (old_mem->mem_type == LOONGGPU_PL_DOORBELL ||
@@ -693,6 +690,7 @@ memcpy:
 		abo->flags &= ~LOONGGPU_GEM_CREATE_CPU_ACCESS_REQUIRED;
 	}
 
+out:
 	/* update statistics */
 	atomic64_add(lg_gbo_to_gem_obj(abo).size, &adev->num_bytes_moved);
 	loonggpu_bo_move_notify(bo, evict, new_mem);
-- 
2.52.0

